File: cli-tools/automation/node/color-matching.js
const axios = require('axios');
const { BASE_URL, getAccessToken } = require('./utils');

async function colorMatching() {
    const imageUrl = process.argv[2];
    const colorMarkers = process.argv[3] ? JSON.parse(process.argv[3]) : [];

    if (!imageUrl || colorMarkers.length === 0) {
        console.log('Usage: node color-matching.js <imageUrl> <colorMarkersJSON>');
        console.log('Example: node color-matching.js "http://example.com/img.jpg" "[{\\"x_coordinate\\":100,\\"y_coordinate\\":100,\\"swatch_color_code\\":\\"#FF5733\\"}]"');
        process.exit(1);
    }

    const token = await getAccessToken();
    try {
        const response = await axios.post(`${BASE_URL}/images/color-matching`, {
            image_url: imageUrl,
            color_markers: colorMarkers
        }, {
            headers: { 'Authorization': `Bearer ${token}` }
        });
        console.log(JSON.stringify(response.data, null, 2));
    } catch (error) {
        console.error('Error:', error.response ? error.response.data : error.message);
    }
}

colorMatching();

File: cli-tools/automation/python/color_matching.py
import sys
import json
import requests
from utils import BASE_URL, get_access_token

def color_matching():
    if len(sys.argv) < 3:
        print('Usage: python color_matching.py <imageUrl> <colorMarkersJSON>')
        sys.exit(1)

    image_url = sys.argv[1]
    try:
        color_markers = json.loads(sys.argv[2])
    except Exception as e:
        print(f"Error parsing JSON: {e}")
        sys.exit(1)

    token = get_access_token()
    headers = {'Authorization': f'Bearer {token}', 'Content-Type': 'application/json'}
    payload = {'image_url': image_url, 'color_markers': color_markers}
    
    try:
        response = requests.post(f"{BASE_URL}/images/color-matching", json=payload, headers=headers)
        response.raise_for_status()
        print(json.dumps(response.json(), indent=2))
    except Exception as e:
        print(f"Error: {e}")

if __name__ == "__main__":
    color_matching()
